src/common/CommandArgs.cpp

```diff
@@ -0,0 +1,61 @@
+#include "CommandArgs.hpp"
+#include "base/Qv2rayBase.hpp"
+
+namespace Qv2ray::common
+{
+    QvCommandArgParser::QvCommandArgParser() : QObject(),
+        noAPIOption("noAPI", QObject::tr("Disable gRPC API subsystems.")),
+        runAsRootOption("I-just-wanna-run-with-root", QObject::tr("Explicitly run Qv2ray as root.")),
+        debugOption("debug", QObject::tr("Enable Debug Output")),
+        withToolbarOption("withToolbarPlugin", QObject::tr("Enable Qv2ray network toolbar plugin")),
+        //
+        helpOption("FAKE"), versionOption("FAKE")
+    {
+        parser.setApplicationDescription(QObject::tr("Qv2ray - A cross-platform Qt frontend for V2ray."));
+        parser.setSingleDashWordOptionMode(QCommandLineParser::ParseAsLongOptions);
+        //
+        parser.addOption(noAPIOption);
+        parser.addOption(runAsRootOption);
+        parser.addOption(debugOption);
+        parser.addOption(withToolbarOption);
+        helpOption = parser.addHelpOption();
+        versionOption = parser.addVersionOption();
+    }
+
+    CommandLineParseResult QvCommandArgParser::ParseCommandLine(QString *errorMessage)
```

Comment:
try make good use of `std::optional<QString>` as return type in C++17.

Reply:
example:
```cpp
std::optional<QString> check() {
  if (something_worked()) return {};
  else return "something is wrong";
}
```

Then use it like this:
```cpp
// when returned nothing, if fails automatically
if (auto errMsg = check()) {
  std::cout << "crap! some error happened: " << errMsg << std::endl;
} else {
  std::cout << "everything is just fine" << std::endl;
}
```

Reply:
This was directly copied from the Qt examples, we need the enums to determine the exit code if the command arg parse failed.